velox/functions/prestosql/HyperLogLogFunctions.h

```diff
@@ -81,4 +80,147 @@ struct EmptyApproxSetWithMaxErrorFunction {
     return true;
   }
 };
+
+class MergeHllFunction : public exec::VectorFunction {
```

Comment:
Does this need to be a vector function? Why can't it be a simple function?

Reply:
I got feedback from the Velox team to use vector function for now as merge_hll requires memory allocation. And that adding Memory Pool to the SimpleFunction API is out of scope, it needs more discussion.

In Vector Functions we can use EvalCtx pool(), compared to a simple function which would have to create it's own memory pool.

Do you think this should be a simple function?

Reply:
@natashasehgal Got it. Have we already created an issue to extend Simple Function API to provide access to a memory pool? If not, let's create one. Let's also update PR description to clarify why we are writing a Vector function and not a simple one.